src/services/notifications.rs

```diff
@@ -57,29 +58,43 @@ pub async fn post_registration(
     }
 
     let chain_id_errors = {
-        let mut output: Vec<&str> = vec![];
-
+        let mut output: HashMap<&str, String> = Default::default();
         for (chain_id, request) in requests.into_iter() {
-            if let Ok(response) = request.await {
-                // tx service errors
-                if let Err(_) = forward_error(response).await {
-                    output.push(chain_id);
+            match request.await {
+                Ok(response) => {
+                    if !response.status().is_success() {
+                        output.insert(
+                            chain_id,
+                            response.text().await.expect("Error response issue"),
```

Comment:
Is this then a json string?

Reply:
Yes, this the `json` of the error returned by the core services.